tsconfig.esm.json

```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./tsconfig.base.json",
+  "compilerOptions": {
+    "module": "ESNext",
+    "moduleResolution": "Node",
```

Comment:
For a published package (or even projects) it's not correct to use any `module` mode other than `nodenext`; that's the only mode that resolves things according to the latest Node.js ESM rules taking into account package `exports` field, imports having to have fully specified paths (including the file extension), etc.

Reply:
Thanks for the feedback.

I've been writing TS for 4+ years now, 'have set up dozens of TS projects.  And yet whenever I have to set up a `tsconfig.json` file it always just feels a bit...

<img src="https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExZ2M0ZnY4aGRsc2g3azhucDVpY2Jhd2I2cTRzcG8zY25rc2hjazJ4cCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/ebITvSXYKNvRm/giphy-downsized.gif" width="300" />

Reply:
https://github.com/uuidjs/uuid/pull/778/files